homeassistant/components/omnilogic/common.py

```diff
@@ -0,0 +1,167 @@
+                    data = get_item_data(single_item, item_kind, current_id, data)
+
+            if "systemId" in item:
+                system_id = item["systemId"]
+                current_id = current_id + (item_kind, system_id)
+                data[current_id] = item
+
+            for kind in ALL_ITEM_KINDS:
+                if kind in item:
+                    data = get_item_data(item[kind], kind, current_id, data)
+
+            return data
+
+        parsed_data = get_item_data(data, "Backyard", (), parsed_data)
+
+        return parsed_data
+
+
+class OmniLogicEntity(CoordinatorEntity):
+    """Defines the base OmniLogic entity."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        kind: str,
+        name: str,
+        item_id: tuple,
+        icon: str,
+    ):
+        """Initialize the OmniLogic Entity."""
+        super().__init__(coordinator)
+
+        bow_id = None
+        entity_data = coordinator.data[item_id]
+
+        backyard_id = item_id[:2]
+        if len(item_id) == 6:
+            bow_id = item_id[:4]
+
+        msp_system_id = coordinator.data[backyard_id]["systemId"]
+        entity_friendly_name = f"{coordinator.data[backyard_id]['BackyardName']} "
+        unique_id = f"{msp_system_id}"
+
+        if bow_id is not None:
+            unique_id = f"{unique_id}_{coordinator.data[bow_id]['systemId']}"
+            entity_friendly_name = (
+                f"{entity_friendly_name}{coordinator.data[bow_id]['Name']} "
+            )
+
+        unique_id = f"{unique_id}_{coordinator.data[item_id]['systemId']}_{kind}"
+
+        if entity_data.get("Name") is not None:
+            entity_friendly_name = f"{entity_friendly_name} {entity_data['Name']}"
+
+        entity_friendly_name = f"{entity_friendly_name} {name}"
+
+        unique_id = unique_id.replace(" ", "_")
+
+        self._kind = kind
+        self._name = entity_friendly_name
+        self._unique_id = unique_id
+        self._item_id = item_id
+        self._icon = icon
+        self._attrs = {}
+        self._msp_system_id = msp_system_id
+        self._backyard_name = coordinator.data[backyard_id]["BackyardName"]
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique, Home Assistant friendly identifier for this entity."""
+        return self._unique_id
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon for the entity."""
+        return self._icon
+
+    @property
+    def device_state_attributes(self):
+        """Return the attributes."""
+        if "Alarms" in self.coordinator.data[self._item_id]:
+            alarms = self.coordinator.data[self._item_id]["Alarms"]
+
+            if len(alarms) != 0:
+                self._attrs["alarm_message"] = alarms[0].get("Message")
+                self._attrs["alarm_comment"] = alarms[0].get("Comment")
```

Comment:
Please give an example of message and comment.

Reply:
There are a couple different examples of alarms, some have only message and some will have a comment:

Message: CHLOR Low Salt
Comment: Salt level is at 2460

or:

Message: pH Timeout

Reply:
Do these alarms represent a state of the device that remains until the alarm is resolved, or may it be an event that happens but doesn't persist?

Eg the pH Timeout, is that a state or an event?

Reply:
They can be a state which remains until alarm is resolved as in low salt or notification to clean the salt cell or they may be an event like the ph timeout. Alarms are used for both types in the Omnilogic system (and unfortunately there isn't a library of them for us to filter out which is which).